### ISSUE TITLE
Question regarding adding custom molecules dataset

### ISSUE THREAD

**Issue by eereenah-fast** [graphdeeplearning/benchmarking-gnns#25]
Hello! Thank you for sharing this project!

I have a question concerning adding new datasets.
 
I have followed the pipeline described in [docs](https://github.com/graphdeeplearning/benchmarking-gnns/blob/master/docs/04_add_dataset.md) to prepare a dataset for molecular regression. My dataset consists of SMILES strings and their respective scores so I'm using [MoleculeCSVDataset ](https://docs.dgl.ai/en/0.4.x/_modules/dgl/data/chem/datasets/csv_dataset.html) method from the DGL library to convert molecules to graphs. I'm also using [CanonicalAtomFeaturizer ](https://docs.dgl.ai/en/0.4.x/generated/dgl.data.chem.CanonicalAtomFeaturizer.html) and [CanonicalBondFeaturizer ](https://docs.dgl.ai/en/0.4.x/generated/dgl.data.chem.CanonicalBondFeaturizer.html) for extracting node and edge features respectively. In the end I get a feature matrix `g.ndata['feat']` of size `n x 74` and `g.edata['feat']` of size `e x 12` for each graph. However, I get shape mismatches during the forward pass. 

Is my understanding of the featurization process correct or should it be done differently? 

Many thanks in advance and apologies for the trivial question.

**Comment by vijaydwivedi75** [comment]
Hi @eereenah-fast, thanks for your interest.

The molecule dataset (ZINC) that we use has label (integer) feature for both node (atom label) and edge (bond label), which are passed to the GNN via an Embedding layer. Maybe check this issue https://github.com/graphdeeplearning/benchmarking-gnns/issues/16.

Therefore, if the features are already one-hot vector, maybe you can use an `nn.Linear()` layer instead of `nn.Embedding()` layer that we have for molecules (ZINC).

Let me know if that doesn't solve!

**Comment by eereenah-fast** [comment]
It worked! Thank you for your help!
